test/integration/noderesourcetopology_cache_test.go

```
@@ -0,0 +1,449 @@
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/client-go/kubernetes"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/scheduler"
+	schedapi "k8s.io/kubernetes/pkg/scheduler/apis/config"
+	fwkruntime "k8s.io/kubernetes/pkg/scheduler/framework/runtime"
+	st "k8s.io/kubernetes/pkg/scheduler/testing"
+
+	topologyv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	"github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/clientset/versioned"
+
+	schedconfig "sigs.k8s.io/scheduler-plugins/apis/config"
+	"sigs.k8s.io/scheduler-plugins/pkg/noderesourcetopology"
+	"sigs.k8s.io/scheduler-plugins/test/util"
+)
+
+const (
+	defaultCacheResyncPeriodSeconds = 5
+	anyNode                         = "*"
+)
+
+type podDesc struct {
+	podName      string
+	isGuaranteed bool
+	isDelete     bool
+	resourcesMap map[string]string
+	expectedNode string
+	// autogenerated
+	pod *corev1.Pod
+}
+
+func (p *podDesc) SetupPod(ns string, initContainer bool) {
+	pt := st.MakePod().Namespace(ns).Name(p.podName)
+	if p.isGuaranteed {
+		p.pod = withLimits(pt, p.resourcesMap, initContainer).Obj()
+	} else {
+		p.pod = withRequests(pt, p.resourcesMap, initContainer).Obj()
+	}
+}
+
+type testCase struct {
+	name                   string
+	nodeResourceTopologies []*topologyv1alpha1.NodeResourceTopology
+	podDescs               []podDesc
+}
+
+func init() {
+	klog.InitFlags(nil)
+}
+
+func TestTopologyCachePluginWithoutUpdates(t *testing.T) {
+	verbose := "0"
+	if val, ok := os.LookupEnv("SCHED_PLUGINS_TEST_VERBOSE"); ok {
+		verbose = val
+	}
+
+	cacheResyncPeriod := int64(defaultCacheResyncPeriodSeconds)
+	if val, ok := os.LookupEnv("SCHED_PLUGINS_TEST_RESYNC_PERIOD_SECS"); ok {
```

Do we provide default values for these environment variables somewhere? Perhaps, we should do that in the Makefile?

---

I'm not sure. In both cases the hardcoded defaults are expected to be good enough; if we want more verbose logs (and these are really useful to debug test failures) we have the option to. The resync period tunable OTOH is probably better to be removed.

---

I don't feel too strongly about this but my only concern was that we have not made it very transparent to someone who wants to run these integration tests that these environment variables exist and can be populated for getting more verbose log or to change the cache resync period. The only way of being aware that these environment variables exist is by inspecting this integration test files. Makefile is typically where environment variables are populated and that is why suggested that.

---

Fair enough, added!